src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/SingletonLocatorAnalyzer.cs

```diff
@@ -88,96 +71,96 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 			}
 		}
 
+		private delegate bool IsSingletonLocatorGet(
+			IMethodSymbol? methodSymbol
+		);
+
 		private delegate bool IsContainerType(
-				ITypeSymbol type,
-				[NotNullWhen( true )] out ITypeSymbol? containedType
-			);
+			ITypeSymbol type,
+			[NotNullWhen( true )] out ITypeSymbol? containedType
+		);
 
 		// Enforce that SingletonLocator can only load actual [Singleton]s
 		private static void EnforceSingletonsOnly(
-			SyntaxNodeAnalysisContext context,
-			Func<INamedTypeSymbol, bool> isSingletonLocator,
+			OperationAnalysisContext context,
+			IMethodSymbol methodSymbol,
+			IsSingletonLocatorGet isSingletonLocatorGet,
 			IsContainerType isContainerType
 		) {
-			var root = GetRootNode( context );
-			if( root == null ) {
+			if( !isSingletonLocatorGet( methodSymbol ) ) {
 				return;
 			}
-			var symbolinfo = context.SemanticModel.GetSymbolInfo( root, context.CancellationToken );
 
-			var method = symbolinfo.Symbol as IMethodSymbol;
+			ITypeSymbol typeArg = methodSymbol.TypeArguments.First();
+			if( isContainerType( typeArg, out ITypeSymbol? containedType ) ) {
+				typeArg = containedType;
+			}
 
-			if( method == null ) {
-				if( symbolinfo.CandidateSymbols == null ) {
-					return;
-				}
+			if( Attributes.Singleton.IsDefined( typeArg ) ) {
+				return;
+			}
 
-				if( symbolinfo.CandidateSymbols.Length != 1 ) {
-					return;
-				}
+			context.ReportDiagnostic( Diagnostic.Create(
+				descriptor: Diagnostics.SingletonLocatorMisuse,
+				location: context.Operation.Syntax.GetLocation(),
+				typeArg.GetFullTypeName()
+			) );
+		}
 
-				//This happens on method groups, such as
-				//  Func<IFoo> fooFunc = OldAndBrokenServiceLocator.Get<IFoo>;
-				method = symbolinfo.CandidateSymbols.First() as IMethodSymbol;
+		private static bool TryGetGetMethod(
+			Compilation compilation,
+			[NotNullWhen( true )] out IMethodSymbol? methodSymbol
+		) {
+			INamedTypeSymbol? locatorType = compilation.GetTypeByMetadataName( "D2L.LP.Extensibility.Activation.Domain.SingletonLocator" );
 
-				if( method == null ) {
-					return;
-				}
+			if( locatorType.IsNullOrErrorType() ) {
+				methodSymbol = null;
+				return false;
 			}
 
-			// At this point method is a non-null IMethodSymbol
-			if( !isSingletonLocator( method.ContainingType ) ) {
-				return;
-
-			}
 
-			if( !IsSingletonGet( method ) ) {
-				return;
+			ImmutableArray<ISymbol> getMemberSymbols = locatorType.GetMembers( "Get" );
+			if( getMemberSymbols.Length != 1 ) {
+				methodSymbol = null;
+				return false;

```

Comment:
Bomb hard here.  If someone adds a member I think it would shutoff the anazlyer

Reply:
Same goes for the methods below

Reply:
Could just emit a diagnostic too saying the analyzer is confused and scared

Reply:
Like if we know we have `SingletonLocator` we should make strong assertions about the rest.

Reply:
Maybe in `RegisterCompilationEndAction`

Reply:
```
	private static bool TryGetGetMethod(
			Compilation compilation,
			[NotNullWhen( true )] out IMethodSymbol? methodSymbol,
			[NotNullWhen( false )] out Diagnostic diagnostic
		);
```

Reply:
But I can also take that on tomorrow.  For 100x improvement, worth it

Reply:
CODEOWNERS sounds more appropriate for that vs. writing more code

Reply:
Given the fragility, I think it's worth it.  Be good to move these into the LMS and check that project references the `D2L.LP` assembly, and if so, assume all the attributes exist

Reply:
Mmmm, yeah a lot of these analyzers really ought to move into the LMS these days.

Reply:
Yeah, I thought of moving ConfigViewer, for instance, but we don't have the testing infrastructure wired up yet.